### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by hamidazim321**
# PR Summary
- In this Project, me and my partner have performed test-driven-development
- We have made tests before coding the methods and have passed all the tests
- We have added the class ```Solver``` and added the methods ```factorial```, ```reverse```, and ```fizzbuzz```

## Updates after previous review
- I have added test for the factorial method when input is 0, the result should equal 1
- I have added test for the scenario when the number is indivisible by both 3 and 5. It should return the string version of that number.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    end
+    it 'Should return buzz for multiple of 5' do
+      result = @solver.fizzbuzz(10)
+      expect(result).to eq('buzz')
+    end
+    it 'Should return fizzbuzz for multiple of both 3 and 5' do
+      result = @solver.fizzbuzz(15)
+      expect(result).to eq('fizzbuzz')
+    end
+  end
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [ ] Kindly add scenario when the number is indivisible by both 3 and 5. It should return the string version of that number.

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  context '#factorial' do
+    it 'Should return 6 as factorial of 3' do
+      factorial = @solver.factorial(3)
+      expect(factorial).to eq(6)
+    end
+    it 'Should raise ArgumentError if a negative integer is given' do
+      expect { @solver.factorial(-1) }.to raise_error(ArgumentError)
+    end
+  end
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [ ] Kindly add the spec for this scenario

  ```
  * When input is 0, the result should equal 1
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @hamidazim321 and @Fombi-Favour,

Good job so far!

## Highlights!
- Good code organization and working specs ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @hamidazim321, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**